stable_baselines3/common/vec_env/base_vec_env.py

```diff
@@ -313,9 +313,12 @@ def __init__(
         venv: VecEnv,
         observation_space: Optional[spaces.Space] = None,
         action_space: Optional[spaces.Space] = None,
-        render_mode: Optional[str] = None,
     ):
         self.venv = venv
+        # Since SB3 v2.x (Gymnasium backend), render_mode attribute is required
+        # and cannot be changed by a wrapper
+        render_mode = venv.render_mode if hasattr(venv, "render_mode") else None
+
```

Comment:
Why didn't you choose to deprecate this argument before removing it?

Reply:
Also, why not moving this line in VecEnv, and also remove render_mode in ``VecEnv.__init__`?

Reply:
this argument was never used so far (only available in SB3 2.x beta) and was buggy anyway.

> Also, why not moving this line in VecEnv, and also remove render_mode in VecEnv.__init__?

where would you move it? in the `render()` method?

Reply:
I guess I would be easier if I just commit my change to show you

Reply:
See 931f09d

Reply:
Tests fails because `DictObsVecEnv` does not implement `get_attr`: the problem lies solely with the test, and it can be easily fixed.

Reply:
that's better indeed =)
but i would do a try/except with a warning for backward compat and for env that don't implement render mode corretly.